src/components/AssertionForm/useAssertionForm.ts

```diff
@@ -32,38 +43,164 @@ export function useAssertionForm() {
     address: USDC as Address,
     enabled: !!USDC,
   });
-  const currenciesData = { weth, dai, usdc };
+
+  const currenciesWithTokenData = useMemo(
+    () => ({ weth, dai, usdc }),
+    [dai, usdc, weth]
+  );
+
+  const currencyDropdownOptions = useMemo(
+    () =>
+      Object.entries(currenciesWithTokenData)
+        .map(([currency, details]) =>
+          !!details
+            ? {
+                value: currency,
+                label: `${details.name} - (${details.symbol})`,
+              }
+            : undefined
+        )
+        .filter(Boolean),
+    [currenciesWithTokenData]
+  );
+
+  const currencyDetails =
+    currenciesWithTokenData[
+      currency?.value as keyof typeof currenciesWithTokenData
+    ];
+  const currencyAddress = currencyDetails?.address;
+  const decimals = currencyDetails?.decimals ?? 18;
+  const currencySymbol = currencyDetails?.symbol ?? "";
+
+  useEffect(() => {
+    if (!!currencyDropdownOptions && !currency) {
+      setCurrency(currencyDropdownOptions[0]);
+    }
+  }, [currencyDropdownOptions, currency]);
+
+  // only weth is supported on goerli, so switch to it if the user is on goerli
+  useEffect(() => {
+    if (chainId === 5) {
+      setCurrency(currencyDropdownOptions[0]);
+    }
+  }, [chainId, currencyDropdownOptions]);
+
+  return useMemo(
+    () => ({
+      currencies: currencyDropdownOptions,
+      currency,
+      setCurrency,
+      currencyAddress,
+      currencyDetails,
+      currencySymbol,
+      decimals,
+    }),
+    [
+      currency,
+      currencyAddress,
+      currencyDetails,
+      currencyDropdownOptions,
+      currencySymbol,
+      decimals,
+    ]
+  );
+}
+
+function useClaim() {
   const [claim, setClaim] = useState("I assert that...");
   const [claimError, setClaimError] = useState("");
-  const [bond, setBond] = useState("1");
-  const [bondInputError, setBondInputError] = useState("");
-  const [bondIsTooLow, setBondIsTooLow] = useState(false);
+
+  useUpdateEffect(() => {
```

Comment:
`useUpdateEffect` is a wrapper around `useEffect` that works the same but it _does not_ run on the first render